client/src/screens/SignUp/SignUp.jsx

```diff
@@ -1,38 +1,157 @@
 import { useState } from 'react';
-import { useNavigate } from 'react-router-dom';
-import { Header } from '../../components/Header/Header.jsx';
-import { Form } from '../../components/Form/Form.jsx';
+import { useNavigate, Link } from 'react-router-dom';
+
 // assets
-import { signUpForm } from '../../services/formData.js';
 import { useDispatch } from 'react-redux';
 import { signUpUser } from '../../services/redux/slices/uiActions.js';
+import { GenericModal } from '../../components/Modal/GenericModal.jsx';
+import './SignUp.scss';
+import { checkEmailAuth, verify } from '../../services/api/users.js';
+import { handleChange } from '../../services/utils/formHandlers';
+import { SingleActionButton } from '../../components/Button/SingleActionButton.jsx';
+import { DoubleActionModal } from '../../components/Modal/DoubleActionModal.jsx';
+
 
 export const SignUp = () => {
   const navigate = useNavigate();
   const dispatch = useDispatch();
+  const [showModal, setShowModal] = useState(false);
+  const [showSuccessModal, setShowSuccessModal] = useState(false);
+  const [modalError, setModalError] = useState("");
+  const [emailError, setEmailError] = useState(null);
   const [newUser, setNewUser] = useState({
-    confirmPassword: '',
+    confirm_password: '',
     email: '',
     first_name: '',
     last_name: '',
     password: '',
   });
 
-  const header = {
-    text: "In order to find projects and connect with other users, we'll need to create an account.",
-    title: 'Register',
-  };
 
   const handleSignUp = async (event) => {
     event.preventDefault();
-    dispatch(signUpUser(newUser));
-    navigate('/');
+    const { email, first_name, last_name, password } = newUser
+    if (email === "" || first_name === "" || last_name === "" || password === "") {
```

Comment:
if we add the `required` attributes to all of our inputs, we can clean up this control flow slightly. HTML will naturally verify that the input has text in it, and if not, will generate a pop-up to prompt the user with `this field cannot be empty` at that specific input, helping them avoid the extra tap of clearing the modal. i liked this initiative though.